test/aqua/operators/test_pauli_sum_op.py

```diff
@@ -0,0 +1,189 @@
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+""" Test PauliSumOp """
+
+import unittest
+from test.aqua import QiskitAquaTestCase
+
+import numpy as np
+from scipy.sparse import csr_matrix
+
+from qiskit import QuantumCircuit, transpile
+from qiskit.aqua.operators import DictStateFn, I, PauliSumOp, SummedOp, X, Y, Z, Zero
+from qiskit.circuit import ParameterVector
+from qiskit.quantum_info import Pauli, SparsePauliOp
+
+
+class TestPauliSumOp(QiskitAquaTestCase):
+    """PauliSumOp tests."""
+
+    def test_construct(self):
+        """ constructor test """
+        sparse_pauli = SparsePauliOp(Pauli(label="XYZX"), coeffs=[2.0])
+        coeff = 3.0
+        pauli_sum = PauliSumOp(sparse_pauli, coeff=coeff)
+        self.assertIsInstance(pauli_sum, PauliSumOp)
+        self.assertEqual(pauli_sum.primitive, sparse_pauli)
+        self.assertEqual(pauli_sum.coeff, coeff)
+        self.assertEqual(pauli_sum.num_qubits, 4)
+
+    def test_add(self):
+        """ add test """
+        pauli_sum = 3 * X + Y
+        self.assertIsInstance(pauli_sum, PauliSumOp)
+
+        expected = PauliSumOp(
+            3.0 * SparsePauliOp(Pauli(label="X")) + SparsePauliOp(Pauli(label="Y"))
+        )
+
+        self.assertEqual(pauli_sum, expected)
+
+    def test_adjoint(self):
+        """ adjoint test """
+        pauli_sum = PauliSumOp(SparsePauliOp(Pauli(label="XYZX"), coeffs=[2]), coeff=3)
+        expected = PauliSumOp(SparsePauliOp(Pauli(label="XYZX")), coeff=-6)
+
+        self.assertEqual(pauli_sum.adjoint(), expected)
+
+        pauli_sum = PauliSumOp(SparsePauliOp(Pauli(label="XYZY"), coeffs=[2]), coeff=3j)
+        expected = PauliSumOp(SparsePauliOp(Pauli(label="XYZY")), coeff=-6j)
+        self.assertEqual(pauli_sum.adjoint(), expected)
+
+    def test_equals(self):
+        """ equality test """
+
+        self.assertNotEqual((X ^ X) + (Y ^ Y), X + Y)
+        self.assertEqual((X ^ X) + (Y ^ Y), (Y ^ Y) + (X ^ X))
+
+        theta = ParameterVector("theta", 2)
+        pauli_sum0 = theta[0] * (X + Z)
+        pauli_sum1 = theta[1] * (X + Z)
+        expected = PauliSumOp(
+            SparsePauliOp(Pauli(label="X")) + SparsePauliOp(Pauli(label="Z")),
+            coeff=1.0 * theta[0],
+        )
+        self.assertEqual(pauli_sum0, expected)
+        self.assertNotEqual(pauli_sum1, expected)
+
+    def test_tensor(self):
+        """ Test for tensor operation """
+        pauli_sum = ((I - Z) ^ (I - Z)) + ((X - Y) ^ (X + Y))
+        expected = (
+            (I ^ I)
+            - (I ^ Z)
+            - (Z ^ I)
+            + (Z ^ Z)
+            + (X ^ X)
+            + (X ^ Y)
+            - (Y ^ X)
+            - (Y ^ Y)
+        )
+        self.assertEqual(pauli_sum, expected)
+
+    def test_permute(self):
+        """ permute test """
+        pauli_sum = PauliSumOp(SparsePauliOp((X ^ Y ^ Z).primitive))
```

Comment:
How about implementing `PauliOp.to_pauli_sum()` to make this line simple?

Reply:
`to_summed_pauli_op()` existed before the commit https://github.com/Qiskit/qiskit-aqua/pull/1440/commits/a22a36cf06270b89b144d700fb7a903db40df970.
I have deleted this because 
- I want users to use PauliSumOp and PauliOp without being aware of class of them.
- I think `to_pauli_sum()` is used in only test. In many case,  X^Y^Z + X^Z^Z and from_list are sufficient.